### ISSUE TITLE
Couldnot get email and email verified claim items in id_token when conformIdTokenClaims is false

### ISSUE THREAD

**Issue by somayaji1983** [panva/node-oidc-provider#1076]
### What happened?

hi @panva 

First of all thanks a lot for such a wonderful library.
I am raising this query / issue after going through FQAs and searching through the lib.

Please find the configurations used 

```

const jwks = require('./jwks.json');
const PORT=3000
const SECURE_KEY="abcd,1234"
const Account = require('./account');

const oidc = new Provider(`https://localhost:3000`, {
  clients: [
    {
      client_id: 'client',
      client_secret: 'client',
      redirect_uris: ['https://jwt.io','https://example.com/'], // using jwt.io as redirect_uri to show the ID Token contents
      response_types: ['id_token','code','code id_token'],
      grant_types: ['implicit','authorization_code','client_credentials','refresh_token'],
      token_endpoint_auth_method: 'client_secret_basic',
    },
  ],
  cookies: {
    keys: SECURE_KEY.split(','),
  },
  pkce: {
    required: () => false,
  },
  scopes: ['openid', 'offline_access', 'email', 'profile'],
  jwks,

  findAccount: Account.findAccount,

  // let's tell oidc-provider you also support the email scope, which will contain email and
  // email_verified claims
  claims: {
    openid: ['sub'],
    address: ['address'], 
    email: ['email', 'email_verified'], 
    phone: ['phone_number', 'phone_number_verified'], 
    profile: ['birthdate', 'family_name', 'gender', 'given_name', 'locale', 'middle_name', 'name', 
      'nickname', 'picture', 'preferred_username', 'profile', 'updated_at', 'website', 'zoneinfo'] 
  },

Account.js
class Account {
  // This interface is required by oidc-provider
  static async findAccount(ctx, id, token) {
    // This would ideally be just a check whether the account is still in your storage
    const account = db.get('users').find({ id }).value();
    if (!account) {
      return undefined;
    }

    return {
      accountId: id,
      // and this claims() method would actually query to retrieve the account claims
      async claims(use,scope,claims,rejected) {
        return {
          sub: id,
          email: account.email,
          email_verified: account.email_verified
        };
      },
    };
  }
```

I am using the following URL to get the code 
 
http://localhost:3000/auth?client_id=azure&response_type=id_token&redirect_uri=https://example.com/&scope=openid email profile&nonce=foobar

and followed by http://localhost:3000/token to obtain the id_token

As per the link [https://github.com/panva/node-oidc-provider/pull/298](url) , when the conformIdTokenClaims = false,
we can provide response_type=code and obtain id_token from /token endpoint and it should be consisting email
and email_verified.

![image](https://user-images.githubusercontent.com/23416571/127763297-14ad0bc0-5cbc-4c92-82de-1c3df70b1650.png)

- The id_token should have email and email_verified claims attributes but they are not.
- If  the response_type=id_token and check token received in url then it consists of both email and email_verified.
- Aslo if specified in authorize url under claims as claims={"id_token":{"email":null}} then we observer email claim is is in the token but not email_verified claim. 

request your inputs on this 

Thanks 
Somayaji

### Version

v7.1.1

### Node.js version

v14.17

### Reproduction & Logs

{
  "sub": "23121d3c-84df-44ac-b458-3d63a9a05497",
  "nonce": "foobar",
  "at_hash": "RYVUktLHLn7Oujjdc8tYOA",
  "aud": "client",
  "exp": 1627807503,
  "iat": 1627803903,
  "iss": "https://localhost:3000"
}

### 

- [X] I have searched the issues tracker and discussions for similar topics and couldn't find anything related.
- [X] I have searched the [FAQ](https://github.com/panva/node-oidc-provider/blob/v7.x/docs/README.md#faq) and couldn't find anything related.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/panva/node-oidc-provider/blob/main/CODE_OF_CONDUCT.md)

**Comment by panva** [comment]
Is it possible your account lookup just doesn't return the claims? I can see an await missing on the DB find query

**Comment by panva** [comment]
For the record I will only look at issues that provide a reproduction repository or a gist I can run in the future. As per the issue form.

**Comment by somayaji1983** [comment]
> Is it possible your account lookup just doesn't return the claims? I can see an await missing on the DB find query

Thanks for your response @panva , here are some more information. I am not using external DB instead it is memory as per the
example that you have provided in the lib. 

Also when I use the same code with response_type=id_token or provide cliams filed in authorize url then I am getting the claims
inside id_token

Can you please provide your inputs on it.

**Comment by panva** [comment]
Eh? Then I misread. You're experiencing the intended behaviour. The default for conformIdTokenClaims is true.

**Comment by somayaji1983** [comment]
> Eh? Then I misread. You're experiencing the intended behaviour. The default for conformIdTokenClaims is true.

I am sorry that I have not posted this part of configuration. I have configured the flag as flase.

```  // if false then id_token in responseType=code request will also have claims
  conformIdTokenClaims: { enabled: false}, 
  features: {
    // disable the packaged interactions
    devInteractions: { enabled: false },
    clientCredentials: { enabled: true },
    revocation: { enabled: true }, 
    introspection: { enabled: true}, 
    claimsParameter: { enabled: true}, 
    userinfo: { enabled: true}, 
  },
});```


but still I am not getting claims when I use response_type=code

**Comment by panva** [comment]
Please understand I'm not going to go around stitching code making something that runs. Please follow the issue form if you believe you've found a bug. It requires to submit a public repository I can run and reproduce your reported behaviour on.

**Comment by panva** [comment]
Nevermind, `conformIdTokenClaims` is enabled&disabled with simple boolean value. It is not under the `features` flag. Your configuration does not actually disable it.

**Comment by panva** [comment]
https://github.com/panva/node-oidc-provider/blob/v7.1.1/docs/README.md#conformidtokenclaims

default: true

**Comment by somayaji1983** [comment]
> https://github.com/panva/node-oidc-provider/blob/v7.1.1/docs/README.md#conformidtokenclaims
> 
> default: true

Thank you @panva , the problem resolved after I modified my configuration as

`conformIdTokenClaims: false`
instead of 
`conformIdTokenClaims: { enabled: false }`

I have followed this issue [https://github.com/panva/node-oidc-provider/issues/608](url) in which you mentioned that it should be an object but not simple Boolean value. 

![image](https://user-images.githubusercontent.com/23416571/127765397-5efd3b85-6427-44fb-ae6c-20f395bef233.png)

Hence I was providing the configuration as an object instead of simple boolean values. 

Thanks a lot for your help.
